opam-check-npm-deps.opam

```diff
@@ -13,20 +13,15 @@ depends: [
   "ocaml" {>= "4.14.0"}
   "reason" {>= "3.8.1"}
   "dune" {>= "3.8"}
-  "opam-client" {>= "2.4" & < "2.5"}
+  "opam-client" {>= "2.5" & < "2.6"}
   "mccs" {>= "1.1+14"}
   "angstrom" {>= "0.15.0"}
   "fmt" {>= "0.9.0"}
   "bos"
   "lwt_ppx"
   "ppx_deriving_yojson"
-  "ppx_expect"
-  "ppx_inline_test"
-  "ppx_let"
-  "ppx_sexp_conv"
   "odoc" {with-doc}
 ]
-available: opam-version >= "2.4" & opam-version < "2.5"
```

Comment:
i think we need this cc @kit-ty-kate 

https://github.com/ocaml/opam-repository/pull/24251#discussion_r1295910443

Reply:
correct. The version of opam and the version of the library should be in sync as much as possible, even more when using `opam-state` and higher-level libraries like `opam-client`